pkg/workceptor/pod.go

```diff
@@ -0,0 +1,72 @@
+package workceptor
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+type KubePodStateHelper interface {
+	GetPodStatus(pod *corev1.Pod) (bool, error)
+	PodHealthy(pod *corev1.Pod, containerName string) (bool, error)
+	PodContainerHealthy(pod *corev1.Pod, containerName string) (bool, error)
+}
+
+// PodContainerHealthy checks if the pod has successfully completed its application logic.
+// this is called after podInfrastructureSuccess has confirmed the pod is in a terminal state.
+func (kw KubeUnit) PodContainerHealthy(pod *corev1.Pod, containerName string) (bool, error) {
+	if pod == nil {
+		return false, fmt.Errorf("pod is nil")
+	}
+
+	var foundContainer *corev1.ContainerStatus = nil
+
+	for i, cs := range pod.Status.ContainerStatuses {
+		if cs.Name == containerName {
+			foundContainer = &pod.Status.ContainerStatuses[i]
+
+			break
+		}
+	}
+	if foundContainer == nil {
+		return false, fmt.Errorf("pod does not contain container %s", containerName)
+	}
+	if foundContainer.State.Terminated == nil { // means it is waiting or running, so application logic has not completed yet. Normal behavior when job completes successfully.
+		return true, nil
+	}
+
+	if foundContainer.State.Terminated.ExitCode != 0 { // exit code of 0 means success
+		return false, fmt.Errorf("container %s exited with code %d: %s", containerName, foundContainer.State.Terminated.ExitCode, foundContainer.State.Terminated.Reason)
+	}
+
+	return true, nil // container terminated with exit code of 0
+}
+
+// PodInfrastructureSuccess checks if the pod has either successfully started, is pending or is running, or has is successfully terminated.
+// Any other state is considered an infrastructure failure.
+func (kw KubeUnit) PodHealthy(pod *corev1.Pod, containerName string) (bool, error) {
+	if pod == nil {
+		return false, fmt.Errorf("pod is nil")
+	}
+
+	containerDiag := fmt.Sprintf("container %s is healthy", containerName)
```

Comment:
Building these error messages is tricky to follow, 
using two different vars, and 6 possible assignments is making me need more coffee 😂 

Can we simplify this please and just create a specific message per return

Reply:
I agree, presetting the containerDiag here is confusing and hard to track what message is in containerDiag when it's used